### DISCUSSION TITLE
problem with field types being lists & mandatory

### DISCUSSION THREAD

**Discussion by Njolf** [prisma/prisma#5663] [Q&A] [answered]
I recently started writing a node.js app with a postgresql database with prisma. i have 3 tables: users, companies, and listings.
the idea is, when you create a user, as that user you can create one or more companies, and then those companies can create one or more listings. for this i figured i would use arrays/lists, with field types being company[] and listing[]. problems, however, arise when you're supposed to create a user or a company, as these lists within them are mandatory, but obviously a user can't have a company with a listing made before they make their account - the way i currently have it, you'd have to make a listing, then a company and then a user, but that can't work either as the company and listing tables have a field labelled owner_id.

So here's my question: how do i get around this problem? i'm genuinely out of ideas on how to implement this (i don't want to limit users to a single company & listing per account).
also, just using company ids and listing ids as Int[] won't help either (again, the company and listing would have to exist first).

**Comment by ryands17** [comment, marked as answer]
@Njolf 👋 

The field types are not mandatory. This will be a many relation when you specify an `[]` and if there are no items, an empty array `[]` will be returned in the response. You do not need to explicitly provide the relation when you create the item. You can read more about relations [here](https://www.prisma.io/docs/concepts/components/prisma-schema/relations). Let me know if this helps :)

**Reply by Njolf** [reply]
Oh, that sounds great! i had assumed that this caused some issues with the returned object not having that field at all, and that it was caused by the query i sent to the database (defining companies as an empty array manually). i'll try experimenting some more with the queries to see if i can get it to work as intended.
Many thanks for the reply!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
